src/meta/raft-store/src/state_machine/snapshot.rs

```diff
@@ -25,3 +29,73 @@ pub struct Snapshot {
     /// The data of the state machine at the time of this snapshot.
     pub data: Vec<u8>,
 }
+
+impl Snapshot {
+    pub fn encode_snapshot_id(last_applied: Option<LogId>, snapshot_index: u64) -> String {
```

Comment:
I prefer using `format_snapshot_id` and `parse_snapshot_id` to indicate that we are building/consuming a `String` instead of a binary.